docs/DeveloperGuide.md

```diff
@@ -1,34 +1,243 @@
+
+#### 3.2.3 Command component
+
+The Command component consists of an abstract `Command` class and the various different commands that inherit the `Command` class.
+Each type of command class (e.g. `addFoodCommand`, `addExerciseCommand` etc) implements an abstract `execute()` method that carries out the command. 
+
+#### 3.2.4 User component
+
+The `User` class contains all information related to the user data (i.e. `name`, `age`, `height`, `weight`
+and `gender`) and handles all operations on these user data (i.e. `getBmi()`)
+
+The `User` class has two constructors and implementations differ based on the constructor used. 
+
+If a `User` object is instantiated without parameters (i.e. `User user = new User()`), the `setup` method in the
+ `user` object will be invoked, allowing users to go through a set of instructions to input their `name`, `age`, `gender`, `height`, `weight` and
+ `fitnessLevel`.
+ 
+If a `User` object is instantiated with defined parameters (i.e. `User user = new User(name, age, height, weight
+, gender)`), the values from the given parameters will be stored within the user class without going through
+ `setup()`.
+
+On startup, the `Fitr` class instantiates a `Storage` object and calls its `loadUserProfile()` method. This method
+returns a `User` type object and is referenced by a pre-declared `User` type variable, which is used throughout the
+running session. 
+
+#### 3.2.5 Storage component
+
+Under the storage component, the `StorageManager` class handles the read and write operations of the various list classes available, such as `ExerciseList` and `FoodList`, through classes such as `ExerciseStorage` and `FoodStorage`.
+
+![StorageManager class diagram](images/StorageClassDiagram.png)
+
+_Figure 3: `StorageManager` class diagram_
+
+Figure 3 above shows the associations and dependencies present with the `StorageManager` class.
+
+On startup, the `Fitr` class creates a new `StorageManager` object. The `StorageManager` class will then create the various storage objects required to load that particular file into the program. 
+This will cause the various storage objects to retrieve the text files from the given file paths and loads the data into the program. 
+If the files do not exist, the various storage objects will create the files. For example, if the file that contains the user's exercise data does not exist, the `ExerciseStorage` object will create a new file to store the user's exercise data.
+
+Each time there is a change in the `FoodList`, `ExerciseList` or `User` objects, or when the program is exiting, `writeExerciseList()`, `writeFoodList()` and `writeUserProfile()` methods in `StorageManager` are invoked. 
+This will save the data onto the respective text files on the hard disk.
+
+#### 3.2.6 Common classes
+
+Classes used by multiple components are in the `fitr.common` package.
+
+## 4. Implementation
+
+This section describes how some of the features in Fitr are implemented.
+
```

Comment:
Implementation part is clearly written, understandable and has good use of diagrams. However, you might want to elaborate more about the functions. For example, you show viewGoal() for 4.3 view, but what are the other stuff that I can view.

Some paragraphs can be wordy, you might want to brak them up into smaller paragraphs.